src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -1095,7 +969,126 @@ class ModelSystem(System):
         """,
     )
 
-    model_system = SubSection(sub_section=SectionProxy('ModelSystem'), repeats=True)
+    total_charge = Quantity(
+        type=np.int32,
+        description="""
+        Total charge of the system.
+        """,
+    )
+
+    total_spin = Quantity(
+        type=np.int32,
+        description="""
+        Total spin state of the system.
+        Not to be confused with the spin multiplicity 2S + 1.
+        """,
+    )
+
+    particle_states = SubSection(
+        section_def=ParticleState.m_def,
+        repeats=True,
+        description='Particle states',
+    )
+
+    sub_systems = SubSection(sub_section=SectionProxy('ModelSystem'), repeats=True)
+
+    def get_chemical_symbols(self, logger: 'BoundLogger') -> list[str]:
+        """
+        Gets the chemical symbols from the particle_states that are AtomsState instances.
+        Args:
+            logger (BoundLogger): The logger to log messages.
+        Returns:
+            list: The list of chemical symbols of the atoms.
+        """
+        chemical_symbols = []
+        for particle_state in self.particle_states:
+            if isinstance(particle_state, AtomsState):
+                # only collect symbols from particle entries that are instances of AtomsState
+                if (
+                    not particle_state.atom_definition_ref
+                    or not particle_state.atom_definition_ref.chemical_symbol
+                ):
+                    logger.warning(
+                        'Could not find `chemical_symbol` in an AtomsState entry.'
+                    )
```

Comment:
In line with the "Dependency Inversion Principle" (SOLI**D**), I'd put the burden on an `AtomsState` superclass to define the interface of providing the representation.
This is then can called for all particle states in a system branch.

```suggestion python
try:
    symbols.append(particle_state.fetch_symbol())
except AttributeError:
    logger.error('Particle cannot provide label information.')
    raise NotImplementedError
```

Note that I modified the name from `chemical_symbols` to `symbols`.
In your discussions with Areas A and C2, what system did you settle on for handling anything that goes outside of the atomic representation?

Reply:
`chemical_symbol` is the element abbreviation and not semantically interchangeable with a generic `symbol` field.
I would rather keep all the logic about "what makes my symbol separate" at this stage. Maybe after `ParticleStates` is finalized.

Reply:
Something like this if there are 2 opinions at the moment, just open an issue with the relevant info (link to this comment) and tag with milestone 1.0 and then you can resolve this without waiting for Nathan's response.

Reply:
did an issue get opened, or should I do it?